src/proxy.rs

```diff
@@ -12,27 +40,43 @@ pub struct ProxyHandlerHttp {
     rewrite: Option<String>,
 }
 
+impl ProxyHandler for ProxyHandlerHttp {
+    fn path(&self) -> &str {
+        self.rewrite.as_ref().map(AsRef::as_ref).unwrap_or_else(|| self.backend.path())
+    }
+
+    fn register(self: Arc<Self>, app: &mut Server<State>) {
+        for method in HTTP_METHODS.iter() {
+            let handler = self.clone();
+            app.at(handler.path()).strip_prefix().method(*method, move |req: Request<State>| {
+                let handler = handler.clone();
+                async move { handler.proxy_request(req).await }
+            });
+        }
```

Comment:
Does make sense to replace the loop + `.method` with [.all](https://docs.rs/tide/0.16.0/tide/struct.Route.html#method.all)?

Reply:
It would, except: https://github.com/thedodd/trunk/issues/95#issuecomment-753508639 (that is what we were doing originally which leg to the bug described in #95) `:)`

Reply:
I see, thanks for the comment.

Then I would recommend to add at least a code comment / `TODO` with the link to the issue and that we should refactor it in the future + probably remove the `HTTP_METHODS` enum.

You can mark the conversation as resolved.

Reply:
Done.